src/module/processcomponents/gtpy_scriptcalculator.cpp

```diff
@@ -33,15 +32,11 @@ GtpyScriptCalculator::GtpyScriptCalculator()
     registerProperty(m_replaceTabBySpaces);
     registerProperty(m_tabSize);
 
-    using PackageInfo = gtpy::package::PackageInfo;
-    gtpy::package::forEachPackage([&](const PackageInfo& pInfo){
-        auto pathProp = new GtObjectPathProperty(
-                    QStringLiteral(""), QStringLiteral(""), QStringLiteral(""),
-                    pInfo.modId, this, QStringList() << pInfo.className);
-        pathProp->hide(true);
+    for (auto* pathProp : qAsConst(m_dynamicPathProps))
+    {
+        pathProp->setParent(this);
```

Comment:
This code seems to be the same in GtPyScriptCalculator and GtPyTask. Can't we push this into the GtpyAbstractScriptComponent constructor?

Reply:
The problem is that GtpyAbstractScriptComponent does not inherit from GtObject, which means we cannot use 'this' to set the parent of the path property inside the GtpyAbstractScriptComponent class. If GtpyAbstractScriptComponent inherited from GtObject, we would run into a diamond inheritance problem because GtpyScriptCalculator inherits from both GtCalculator and GtpyAbstractScriptComponent. To avoid a larger code change, I decided to call setParent in both GtpyScriptCalculator and GtpyTask.

This approach acts more like a fallback to prevent memory leaks, as the destructor of GtpyAbstractScriptComponent deletes all pointers from m_dynamicPathProps as well.

Do you know of another way to handle this problem?

Reply:
In this case, you could add a `GtpyAbstractScriptComponent::init(GtObject*)` method that you could call from the constructors, which contains the common code.

Reply:
To set the parent, `GtpyAbstractScriptComponent::init(GtObject*) `would work. However, `GtObject::registerProperty(GtAbstractProperty&)` is a protected method, so we cannot register the property by passing the GtObject instance to `GtpyAbstractScriptComponent`.